crates/storage/provider/src/providers/snapshot/writer.rs

```diff
@@ -97,7 +92,13 @@ impl<'a> SnapshotProviderRW<'a> {
         // Commits offsets and new user_header to disk
         self.writer.commit()?;
 
-        self.reader.update_index(self.writer.user_header().segment(), self.get_max_block())?;
+        if let Some(metrics) = &self.metrics {
+            metrics.record_segment_operation(
+                self.writer.user_header().segment(),
+                SnapshotProviderOperation::CommitWriter,
+                Some(start.elapsed()),
+            );
+        }
```

Comment:
nit: move below update so if it fails we don't report the metric

Reply:
whether we index correctly or not, the commit itself did go through so i think its a valid metric